sql/core/src/test/scala/org/apache/spark/sql/connector/AlterTableTests.scala

```diff
@@ -1141,6 +1141,36 @@ trait AlterTableTests extends SharedSparkSession {
     }
   }
 
+  test("AlterTable: remove nonexistent table property") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id int) USING $v2Format TBLPROPERTIES('test' = '34')")
+
+      val tableName = fullTableName(t)
+      val table = getTableMetadata(tableName)
+
+      assert(table.name === tableName)
+      assert(table.properties ===
+        withDefaultOwnership(Map("provider" -> v2Format, "test" -> "34")).asJava)
+
+      val exc = intercept[AnalysisException] {
+        sql(s"ALTER TABLE $t UNSET TBLPROPERTIES ('unknown')")
+      }
+      assert(exc.getMessage.contains("Attempted to unset non-existent property 'unknown'"))
+
+      // Reserved property "comment" should be allowed regardless.
```

Comment:
even if the table has no comment?

Reply:
Yes, that's the current behavior of V1 command:
https://github.com/apache/spark/blob/3b26bc25362a245a610c3e222b971b4ae612bc3e/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L307-L314

Since table comment is optional, it will just be set to `None` if the table had no comment:
https://github.com/apache/spark/blob/3b26bc25362a245a610c3e222b971b4ae612bc3e/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L316